### PR TITLE
Disable the colors when stdout isn't a tty

### PR DESCRIPTION
**Pull request by thufschmitt**
Fix https://github.com/tweag/nickel/issues/1500

~The test is probably wrong. I'll try and make it correct, but if there's a better way of writing it, please let me know.~

### CODE REVIEW

**Comment by yannham** [comment]
FYI, clippy is complaining. As a oneliner, you can run:

```console
cargo clippy --all-targets --workspace -- --deny warnings --allow clippy::new-without-default --allow clippy::match_like_matches_macro
```

to reproduce what the CI is doing. We discussed just today that the developer environment didn't give proper access to clippy, because it's not in the pre-commit hooks anymore :sweat_smile: that should be fixed, but in the meantime, this is the script I use

File: cli/tests/integration/main.rs [hunk 1]
```diff
@@ -44,3 +44,27 @@ fn doc_creates_output_files() {
...
+    let nickel_bin = env!("CARGO_BIN_EXE_nickel");
+    let mut nickel = Command::new(nickel_bin)
+        .args(&["export"])
+        .stdin(Stdio::piped())
+        .stdout(Stdio::piped())
+        .spawn()
+        .expect("Nickel should be runnable");
+    let Some(mut stdin) = nickel.stdin.take() else {
+            panic!("couldn't retrieve stdin handle to Nickel")
+        };
```
[the remark this answers is not in the document]
Comment (thufschmitt) [reply cli/tests/integration/main.rs]: Indeed. I just copied the test above, but `expect` feels more natural here

**Comment by thufschmitt** [comment]
Fixed the clippy warning, and made the test actually test something (by grepping for the correct string and in the right stream). Should be much better now

**Review by vkleen — APPROVED** [review]

File: cli/tests/integration/main.rs [hunk 2]
```diff
@@ -44,3 +44,37 @@ fn doc_creates_output_files() {
...
+    stdin
+        .write_all(b"1+{}")
+        .expect("writing into Nickel stdin should work");
+    drop(stdin);
+    let output = nickel
+        .wait_with_output()
+        .expect("couldn't retrieve stdout handle to Nickel");
+    for raw_stream in [output.stdout, output.stderr] {
+        let stream =
+            String::from_utf8(raw_stream).expect("The result of Nickel should be valide utf8");
```
Comment (yannham) [inline cli/tests/integration/main.rs:71]:
```suggestion
            String::from_utf8(raw_stream).expect("The result of Nickel should be valid utf8");
```

**Review by yannham — APPROVED** [review]

**Pull request merged**